android/src/main/java/com/it_nomads/fluttersecurestorage/FlutterSecureStoragePlugin.java

```diff
@@ -64,50 +84,9 @@ public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
     }
 
     @Override
-    public void onMethodCall(MethodCall call, Result result) {
-        try {
-            switch (call.method) {
-                case "write": {
-                    String key = getKeyFromCall(call);
-                    Map arguments = (Map) call.arguments;
-
-                    String value = (String) arguments.get("value");
-                    write(key, value);
-                    result.success(null);
-                    break;
-                }
-                case "read": {
-                    String key = getKeyFromCall(call);
-
-                    String value = read(key);
-                    result.success(value);
-                    break;
-                }
-                case "readAll": {
-                    Map<String, String> value = readAll();
-                    result.success(value);
-                    break;
-                }
-                case "delete": {
-                    String key = getKeyFromCall(call);
-
-                    delete(key);
-                    result.success(null);
-                    break;
-                }
-                case "deleteAll": {
-                    deleteAll();
-                    result.success(null);
-                    break;
-                }
-                default:
-                    result.notImplemented();
-                    break;
-            }
-
-        } catch (Exception e) {
-            result.error("Exception encountered", call.method, e);
-        }
+    public void onMethodCall(@NonNull MethodCall call, @NonNull Result rawResult) {
+        MethodResultWrapper result = new MethodResultWrapper(rawResult);
+        new Thread(new MethodRunner(call, result)).start();
```

Comment:
How heavy is the creation of a thread for each call? Are there any risks?

Reply:
I think it would depend on your app workload. You could use an ExecutorService to keep a thread in a pool for you if it will be called often. If the secure storage methods were called repeatedly, like in a benchmarking scenario, a thread pool would probably perform much better. However, my app is light on secure storage calls so holding a thread that is rarely used in the pool probably wouldn't be a net gain.